upi/openstack/network.yaml

```diff
@@ -62,18 +64,18 @@
       - subnet: "{{ os_subnet }}"
         ip_address: "{{ os_apiVIP }}"
     register: _api_ports
-    when: os_subnet6 is not defined
+    when: os_subnet6_range is not defined
 
   - set_fact:
       api_ports: "{{ _api_ports }}"
-    when: _api_ports.changed
+    when: not (_api_ports.skipped|default(false)) and not _api_ports.failed
```

Comment:
something I just noticed, maybe we don't need to `set_fact` here and also at line 82.

Reply:
We do. With registering `_api_ports` (and `_ingress_ports`) for the case with single stack, both registered variables in the lines 59-67 (for API port) will be overwritten on lines 73-78 even if this task will be skipped. So, we need to persist value as a fact.